frontend/src/templates.ts

```diff
@@ -25,8 +30,15 @@ export const templates = {
             </div>
             <div>
               <div class="aa-ItemTitle">
-                ${title}
+                ${hierarchy?.lvl0 ?? title}
```

Comment:
lvl0 is more the "meta category" than the title, I would not replace the title.
For example the vuejs doc:
"cookbook beta" is the lvl0
<img width="590" alt="Screenshot 2020-12-07 at 14 18 49" src="https://user-images.githubusercontent.com/1637651/101355673-311af700-3897-11eb-962f-39bec64c4019.png">

Reply:
Our lvl 0 actually corresponds to the lvl1 of the Docsearch Vuejs index. They are indeed able to get the "meta category" from the sidebar because they do custom extraction, but on our side we only use the main content.

Currently in 100% of our example pages, our lvl0 (i.e. `h1`) **is** the title.
Showing title + lvl0 will duplicate the title.

Of course for pages where the `h1` is not the title, or if there is a second `h1`, the UI won't display the title of the page, but that doesn't mean it won't be relevant. We could also check the page title against the lvl0 to decide if we display the title or not.

Reply:
we should probably have followed the same rule and put h1 in lvl1 to avoid confusion

Reply:
We still can, but we would have an always-null `lvl0` 😕 
Happy to rediscuss it

Reply:
it's really up to us, always putting h1 in lvl0 is already deviating from docsearch schema (n.b: lvl0 is not a mandatory field in docsearch). Some configs are putting h1 in it, some are empty, some are static, etc...

If on our case it's always the h1, we can keep it like this it won't break anything, but it's easier to map Heading to the level (h1 = lvl1, h2 = lvl2, etc...). However it could displays "incorrectly" when using docsearch UI

Reply:
It's also for consistency with the `hierarchicalCategories` field, meant to be compatible with InstantSearch: https://www.algolia.com/doc/api-reference/widgets/hierarchical-menu/react/
According to the doc, `lvl0` is mandatory.

Reply:
okay let's focus on instantsearch then 👌🏻